pkg/syncer/metadatasyncer.go

```diff
@@ -397,21 +415,46 @@ func pvUpdated(oldObj, newObj interface{}, metadataSyncer *metadataSyncInformer)
 	}
 	log.Debugf("PVUpdated: PV Updated from %+v to %+v", oldPv, newPv)
 
-	// Verify if pv is a vsphere csi volume
-	if oldPv.Spec.CSI == nil || newPv.Spec.CSI == nil || newPv.Spec.CSI.Driver != csitypes.Name {
-		log.Debugf("PVUpdated: PV is not a Vsphere CSI Volume: %+v", newPv)
-		return
-	}
 	// Return if new PV status is Pending or Failed
 	if newPv.Status.Phase == v1.VolumePending || newPv.Status.Phase == v1.VolumeFailed {
 		log.Debugf("PVUpdated: PV %s metadata is not updated since updated PV is in phase %s", newPv.Name, newPv.Status.Phase)
 		return
 	}
-	// Return if labels are unchanged
-	if (oldPv.Status.Phase == v1.VolumeAvailable || oldPv.Status.Phase == v1.VolumeBound) && reflect.DeepEqual(newPv.GetLabels(), oldPv.GetLabels()) {
-		log.Debugf("PVUpdated: PV labels have not changed")
-		return
+	// Verify if csi migration is ON and check if there is any label update or migrated-to annotation was received for the PV
+	if metadataSyncer.configInfo.Cfg.FeatureStates.CSIMigration && newPv.Spec.VsphereVolume != nil {
+		if oldPv.Status.Phase == v1.VolumeBound &&
```

Comment:
I think we should also consider oldPv.Status.Phase == v1.VolumeAvailable

```
 (oldPv.Status.Phase == v1.VolumeAvailable || oldPv.Status.Phase == v1.VolumeBound)
```

VolumeAvailable will the state of the PV statically created using an in-tree volume path.